keps/sig-cloud-provider/20191004-out-of-tree-credential-providers.md

```diff
@@ -185,16 +184,23 @@ kind: credentialprovider
 apiVersion: v1alpha1
 providers:
 -
+  name: ecr-creds
   imageMatchers:
   - *.dkr.ecr.*.amazonaws.com
   - *.dkr.ecr.*.amazonaws.com.cn
-  exec:
-    command: ecr-creds
-    args: token
-    apiVersion: v1alpha1
+  extraArgs:
+    region: us-west-2
 ```
 
-Where ecr-creds is a binary that vends ecr credentials.  This would execute the binary `ecr-creds` with the argument `token` for the image `012345678910.dkr.ecr.us-east-1.amazonaws.com/my-image`.
+Where ecr-creds is a binary that vends ecr credentials.  This would execute the binary `ecr-creds` for the image `012345678910.dkr.ecr.us-east-1.amazonaws.com/my-image`.  The extra args would be passed in as environment variables, so the entire command would be:
+
+```
+REGION=us-west-2 ecr-creds get-credentials
```

Comment:
Wondering if passing this as argument makes more sense:
```bash
$ ecr-creds get-credentials region=us-west2,foo=bar,...    # or something similar
```

I think there could be unintended consequences of using env vars (e.g. SDK also checks the same env var but could behave differently from what the plugin may want to do).